src/transport/TcpRemotingClient.cpp

```diff
@@ -319,8 +319,15 @@ std::shared_ptr<TcpTransport> TcpRemotingClient::CreateTransport(const string& a
         if (connectStatus == TCP_CONNECT_STATUS_SUCCESS) {
           return tcp;
         } else if (connectStatus == TCP_CONNECT_STATUS_WAIT) {
-          std::shared_ptr<TcpTransport> pTcp;
-          return pTcp;
+          connectStatus = tcp->waitTcpConnectEvent(static_cast<int>(m_tcpConnectTimeout));
```

Comment:
don't wait event during locking.

Reply:
Any suggustions? waitTcpConnectEvent is also called in the funtion if connection not in table

Reply:
fixed.